stdlib/3/tokenize.pyi

```diff
@@ -2,34 +2,44 @@
 #
 # NOTE: This dynamically typed stub was automatically generated by stubgen.
 
-from typing import Any, Union, TextIO
+from typing import Any, Callable, Generator, Iterable, List, NamedTuple, Optional, Union, Sequence, TextIO, Tuple
 from builtins import open as _builtin_open
 from token import *  # noqa: F403
 
-COMMENT = ...  # type: Any
-NL = ...  # type: Any
-ENCODING = ...  # type: Any
+COMMENT = 0
+NL = 0
+ENCODING = 0
 
-class TokenInfo:
+_Position = Tuple[int, int]
+
+class TokenInfo(NamedTuple):
+    type: int
+    string: str
+    start: _Position
+    end: _Position
+    line: str
     @property
-    def exact_type(self): ...
+    def exact_type(self) -> int: ...
+
+# Backwards compatible tokens can be sequences of a shorter length too
+_Token = Union[TokenInfo, Sequence[Union[int, str, _Position]]]
 
 class TokenError(Exception): ...
 class StopTokenizing(Exception): ...
 
 class Untokenizer:
-    tokens = ...  # type: Any
-    prev_row = ...  # type: Any
-    prev_col = ...  # type: Any
-    encoding = ...  # type: Any
-    def __init__(self): ...
-    def add_whitespace(self, start): ...
-    def untokenize(self, iterable): ...
-    def compat(self, token, iterable): ...
+    tokens = ...  # type: List[str]
+    prev_row = ...  # type: int
+    prev_col = ...  # type: int
+    encoding = ...  # type: Optional[str]
+    def __init__(self) -> None: ...
+    def add_whitespace(self, start: _Position) -> None: ...
+    def untokenize(self, iterable: Iterable[_Token]) -> str: ...
+    def compat(self, token: Sequence[Union[int, str]], iterable: Iterable[_Token]) -> None: ...
 
-def untokenize(iterable): ...
-def detect_encoding(readline): ...
-def tokenize(readline): ...
+def untokenize(iterable: Iterable[_Token]) -> Union[str, bytes]: ...
```

Comment:
We should almost never return a Union because Union handling is often buggy in mypy and using Unions forces calling code into tedious isinstance checks. I think this one should just return Any.